timbuctoo-instancev4/src/main/java/nl/knaw/huygens/timbuctoo/server/databasemigration/ScaffoldMigrator.java

```diff
@@ -19,99 +25,125 @@
-                  .withProperty("date", localProperty("document_date", datable))
-
-              )
-              .withCollection("concepts", collection ->
-                collection
-                  .withDisplayName(localProperty("label"))
-              )
-              .withCollection("relations", CollectionBuilder::isRelationCollection))
-          .build();
-        db.initDb(
-          mappings,
-          relationType("person", "hasBirthPlace", "location", "isBirthPlaceOf", false, false, false, UUID.randomUUID()),
-          relationType("person", "hasDeathPlace", "location", "isDeathPlaceOf", false, false, false, UUID.randomUUID()),
-          relationType("collective", "hasMember", "person", "isMemberOf", false, false, false, UUID.randomUUID()),
-          relationType("collective", "locatedAt", "location", "isHomeOf", false, false, false, UUID.randomUUID()),
-          relationType("document", "isCreatedBy", "person", "isCreatorOf", false, false, false, UUID.randomUUID()),
-
-          // TODO+FIXME, these BIA relationTypes should be made VRE specific and editable before mapping
-          // person to person relations
-          relationType("concept", "hasFirstPerson", "person", "isFirstPersonInRelation",
-            false, false, false, UUID.randomUUID()),
-          relationType("concept", "hasSecondPerson", "person", "isSecondPersonInRelation",
-            false, false, false, UUID.randomUUID()),
-          relationType("concept", "hasPersonToPersonRelationType", "concept", "isPersonToPersonRelationTypeOf",
-            false, false, false, UUID.randomUUID()),
-
-          // states of persons
-          relationType("concept", "hasStateType", "concept", "isStateTypeOf", false, false, false, UUID.randomUUID()),
-          relationType("concept", "isStateOfPerson", "person", "hasPersonState",
-            false, false, false, UUID.randomUUID()),
-          relationType("concept", "isStateLinkedToInstitute", "collective", "isInstituteLinkedToState",
-            false, false, false, UUID.randomUUID()),
-          relationType("concept", "isStateLinkedToLocation", "location", "isLocationLinkedToState",
-            false, false, false, UUID.randomUUID()),
-
-          // data lines for persons
-          relationType("concept", "hasDataLineType", "concept", "isDataLineTypeOf",
-            false, false, false, UUID.randomUUID()),
-          relationType("concept", "isDataLineForPerson", "person", "hasDataLine",
-            false, false, false, UUID.randomUUID()),
-
-          // scientist_bios for persons
-          relationType("concept", "hasFieldOfInterest", "concept", "isFieldOfInterestOf",
-            false, false, false, UUID.randomUUID()),
-          relationType("concept", "isScientistBioOf", "person", "hasScientistBio",
-            false, false, false, UUID.randomUUID())
-        );
-      }
-      return TransactionState.commit();
-    });
+    DataStoreOperations db = new DataStoreOperations(
```

Comment:
this is a rather specific constructor that only works under specific DataStoreOperations conditions. Perhaps make this a package private static method that has a descriptive name? (migrationDataStoreOperations, or indexAndChangeListenerAvoidingDataStoreOperations or something like that